src/Discord/Internal/Types/Interactions.hs

```diff
@@ -449,6 +449,10 @@ data OptionDataValue
       { optionDataValueName :: T.Text,
         optionDataValueNumber :: Either T.Text Number
       }
+  | OptionDataValueAttachment
+      { optionDataValueName :: T.Text,
+        optionDataValueAttachment :: Snowflake
```

Comment:
I think what you were looking at is when you add dropdown choices for a command application type.
As far as I understood the source code `OptionDataValue` family of types are what is sent back as a response from the slash commands in Discord. While the `OptionValue` family of types are what would be on the Discord UI side, and here is where the value would be either `string, integer, or double` if the command accepted choices.

At [Data Option Structure](https://discord.com/developers/docs/interactions/receiving-and-responding#interaction-object-application-command-interaction-data-option-structure) we see that value is either `string, integer, double, or boolean`. But during my testing the *string* was always a `Snowflake` string.

I can test this further or change it to `String` to match the docs. I'm not sure how to test this further thou.

Reply:
I can't imagine it wouldn't be a snowflake; I think it would be worth adding something about the below to a comment here

> And the Resolved Data structure that is sent back with the response is a Map of Snowflakes to [attachment](https://discord.com/developers/docs/resources/channel#attachment-object) objects. Hence why I chose the type to be Snowflake as well.

If you can think of some testing that would be appreciated, but if you can't lmk and I'll stamp this through.

Reply:
When you say to add a comment "*here*", are you referring to haskell comment in code or some sort of github comment somewhere? And if I were to add some haskell comment should it be a doc comment or regular?

I can't think of a good way to test this other than having Debug.Trace on the client side and check it manually. I think you can stamp this through.

Reply:
A doc comment on the field saying why it's a snowflake instead of some other data type would be good